src/main/java/com/google/edith/servlets/UserInsights.java

```diff
@@ -98,54 +103,42 @@ public void updateUserStats(List<Key> newItems) {
    * @return A map relating a time period to the spending in that time period.
    */
   public Map<String, String> aggregateUserData() { 
-    Entity userStats = retreiveUserStats();
-    List<Key> items = (List<Key>) userStats.getProperty("Items");
-
-    items.sort(SORT_BY_DATE);
-
+    List<Entity> items = retreiveUserItems();
+    if (items == null || items.isEmpty()) {
+        return null;
+    }
     return calculateWeeklyTotal(items);
   }
 
   /**
-   * Creates a String-Intger map relating weekly periods to spending.
+   * Creates a String-String map relating weekly periods to spending.
    * @param items - A list of {@code Key} objects that reference Item entities
                   - in the datastore.
    * @return Creates a map with keys for each ending day of a weekly period and  
    *         values for the total spending during that period. 
    */
-  public Map<String, String> calculateWeeklyTotal(List<Key> items)  {
+  public Map<String, String> calculateWeeklyTotal(List<Entity> items)  {
     Map<String, String> weeklyTotals = new HashMap<String, String>();
-    LocalDate currentEndOfWeek;
-    try {
-      currentEndOfWeek = getEndOfWeek(LocalDate.parse((String) datastore.get(items.get(0))
-                                          .getProperty("date"), dateFormatter));
-    } catch(EntityNotFoundException e) {
-      System.err.println("Error: Entity could not be located");
-      return null;
-    }
+    LocalDate currentEndOfWeek = getEndOfWeek(LocalDate.parse((String) 
+                            items.get(0).getProperty("date"), dateFormatter));
     double weeklyTotal = 0;
     Entity itemEntity;
-    for(Key item : items) {
-      try {
-        itemEntity = datastore.get(item);
-        LocalDate itemDate = LocalDate.parse((String) datastore.get(item).getProperty("date"),
+    for(Entity item : items) {
+      LocalDate itemDate = LocalDate.parse((String) item.getProperty("date"),
                                              dateFormatter);
-        if(ChronoUnit.DAYS.between(itemDate, currentEndOfWeek) < 0) {
-          weeklyTotals.put(currentEndOfWeek.toString(), Double.toString(weeklyTotal));
-          currentEndOfWeek = getEndOfWeek(itemDate);
-          weeklyTotal = 0;
-        }
+      
+      if(ChronoUnit.DAYS.between(itemDate, currentEndOfWeek) < 0) {
```

Comment:
I _think_ this might not be syncd to the latest changes? I feel like we updated this so it reads a little easier (instead of negative time)?

Reply:
Oh no, this is a different version from a different branch. I edited it to how we have it on the master branch.